OgreMain/include/OgreSceneLoaderManager.h

```diff
@@ -0,0 +1,50 @@
+#ifndef __SCENELOADERMANAGER_H__
+#define __SCENELOADERMANAGER_H__
+
+#include "OgreSingleton.h"
+#include "OgreSceneLoader.h"
+
+namespace Ogre {
+    
+    /** \addtogroup Core
+    *  @{
+    */
+    /** \addtogroup Resources
+    *  @{
+    */
+    /** This class defines an interface for registering and using SceneLoaders.
+    */
+    class _OgreExport SceneLoaderManager : public Singleton<SceneLoaderManager>
+    {
+    public:
+        SceneLoaderManager();
+        virtual ~SceneLoaderManager();
+        /** Register a new SceneLoader
+            @param name The name for the SceneLoader for lookup.
+            @param sl Pointer to the SceneLoader instance.
+        */
+        void registerSceneLoader(const String& name, SceneLoader *sl);
+        /** Load a scene from a SceneLoader
+        @remarks This is usually equivalent to calling
+            SceneLoader::load(stream, groupName, rootNode)
+        @param loaderName The name of the SceneLoader that will be used for loading.
+        @param stream Weak reference to a data stream which is the source of the scene
+        @param groupName The name of a resource group which should be used if any resources
+            are created during the parse of this script.
+        @param rootNode The root node for the scene being created
+        */
+        void load(const String& loaderName, DataStreamPtr& stream, const String& groupName, SceneNode *rootNode);
```

Comment:
instead of explicitly requiring a loader name, you can use `stream->getName()` to inspect the file extension and automatically select the loader based on that

Reply:
Should there still be a check that all extensions supported by the SceneLoader being added have not yet been registered by another SceneLoader and that the extensions don't repeat in the same SceneLoader? It seems unnecessary now that we don't query by name. (So there can be overlaps and such, but the first one registered gets priority.)

Reply:
yes, people might register common extensions like "xml" that could map to different incompatible file formats.

Reply:
Will warn unless only one format is specified.